homeassistant/components/yale_smart_alarm/entity.py

```diff
@@ -0,0 +1,26 @@
+"""Base class for yale_smart_alarm entity."""
+
+from homeassistant.const import CONF_USERNAME
+from homeassistant.helpers.entity import DeviceInfo, Entity
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER, MODEL
+from .coordinator import YaleDataUpdateCoordinator
+
+
+class YaleEntity(CoordinatorEntity, Entity):
+    """Base implementation for Yale device."""
+
+    def __init__(self, coordinator: YaleDataUpdateCoordinator, data: dict) -> None:
+        """Initialize an Yale device."""
+        super().__init__(coordinator)
+        self._coordinator = coordinator
```

Comment:
```suggestion
```

You can drop this and use `self.coordinator` everywhere since `CoordinatorEntity` will set it

Reply:
When I do this I can no longer reach yale as in `self.coordinator.yale` which is defined in coordinator.py
But I don't understand why and therefore left this in here to set it explicit. using `self._coordinator`

Reply:
Ok, got it but can't remove it unless I change the lock and alarm panel at the same time so will submit it for new PR, or we just do all of it in this PR?

Reply:
It's fine to do here